packages/circuits/circuits/proveGraffitiPreimage.circom

```diff
@@ -0,0 +1,61 @@
+/*
+    Prove:
+    1. User exists in the Global State tree
+    2. User own the epoch key
+    3. Pre-image of graffiti
+*/
+
+include "../node_modules/circomlib/circuits/comparators.circom";
+include "../node_modules/circomlib/circuits/gates.circom";
+include "../node_modules/circomlib/circuits/mux1.circom";
+include "../node_modules/circomlib/circuits/poseidon.circom";
+include "./sparseMerkleTree.circom";
+
+template proveGraffitiPreimage(GST_tree_depth, user_state_tree_depth, epoch_tree_depth, EPOCH_KEY_NONCE_PER_EPOCH) {
+
+    /* 1. Check if yauser exists in the Global State Tree and verify epoch key */
+    component verify_epoch_key = VerifyEpochKey(GST_tree_depth, epoch_tree_depth, EPOCH_KEY_NONCE_PER_EPOCH);
```

Comment:
and there is no `VerifyEpochKey` in unirep social repo.
So you can either copy-paste the lines in `ProveNegativeReputation` or `ProveSubsidyKey` or you can write another template `VerifyEpochKey` to verify epoch key.